diff --git a/CMakeLists.txt b/CMakeLists.txt
new file mode 100644
index 0000000..79a8f51
--- /dev/null
+++ b/CMakeLists.txt
@@ -0,0 +1,66 @@
+cmake_minimum_required(VERSION 3.10)
+project(io500)
+
+include_directories(include src test)
+link_directories(lib)
+
+set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -g -O2")
+set(LIBS
+  m daos daos_common dfs gurt uuid
+  mfu_dfind mfu mpi
+)
+
+set(IOR_VERSION \"io500-isc21\")
+add_compile_options("-DVERSION=${IOR_VERSION}")
+
+add_library(io500
+  src/util.c
+  src/ini-parse.c
+  src/phase_dbg.c
+  src/phase_opt.c
+  src/phase_timestamp.c
+  src/phase_find.c
+  src/phase_find_easy.c
+  src/phase_find_hard.c
+
+  src/phase_ior_easy.c
+  src/phase_ior_easy_read.c
+  src/phase_ior.c
+  src/phase_ior_easy_write.c
+  src/phase_ior_hard.c
+  src/phase_ior_hard_read.c
+  src/phase_ior_hard_write.c
+  src/phase_ior_rnd.c
+  src/phase_ior_rnd_write.c
+  src/phase_ior_rnd_read.c
+
+  src/phase_mdtest.c
+  src/phase_mdtest_easy.c
+  src/phase_mdtest_easy_delete.c
+  src/phase_mdtest_easy_stat.c
+  src/phase_mdtest_easy_write.c
+  src/phase_mdtest_hard.c
+  src/phase_mdtest_hard_delete.c
+  src/phase_mdtest_hard_read.c
+  src/phase_mdtest_hard_stat.c
+  src/phase_mdtest_hard_write.c
+  src/phase_mdworkbench.c
+  src/phase_mdworkbench_create.c
+  src/phase_mdworkbench_delete.c
+  src/phase_mdworkbench_bench.c
+)
+target_link_libraries(io500 pfind lz4 ${LIBS})
+
+add_executable(io500-bin src/main.c)
+set_target_properties(io500-bin PROPERTIES OUTPUT_NAME io500)
+add_dependencies(io500-bin io500)
+target_link_libraries(io500-bin io500 aiori ${LIBS})
+
+add_executable(io500-verifier src/verifier.c)
+add_dependencies(io500-verifier io500)
+target_link_libraries(io500-verifier io500 ${LIBS})
+
+set(CMAKE_INSTALL_INCLUDEDIR ${CMAKE_INSTALL_PREFIX}/include)
+install(TARGETS io500 DESTINATION ${CMAKE_INSTALL_LIBDIR})
+install(TARGETS io500-bin DESTINATION ${CMAKE_INSTALL_BINDIR})
+install(TARGETS io500-verifier DESTINATION ${CMAKE_INSTALL_BINDIR})
